packages/jest-serializer/src/index.ts

```diff
@@ -125,21 +125,40 @@ export function serialize(content: unknown): Buffer {
 
 // Synchronous filesystem functions.
 
-export function readFileSync(filePath: Path): any {
-  return v8.deserialize
-    ? v8.deserialize(fs.readFileSync(filePath))
-    : jsonParse(fs.readFileSync(filePath, 'utf8'));
+export async function readFile<T = unknown>(filePath: Path): Promise<T> {
+  return new Promise<T>((resolve, reject) => {
+    fs.readFile(filePath, (err, data) => {
+      try {
+        if (err) {
+          throw err;
+        }
+        resolve(deserialize<T>(data));
+      } catch (ex) {
+        reject(ex);
+      }
+    });
+  });
 }
 
-export function writeFileSync(filePath: Path, content: any) {
-  return v8.serialize
-    ? fs.writeFileSync(filePath, v8.serialize(content))
-    : fs.writeFileSync(filePath, jsonStringify(content), 'utf8');
+export function writeFile(filePath: Path, content: any): Promise<void> {
+  return new Promise<void>((resolve, reject) => {
+    try {
+      fs.writeFile(filePath, serialize(content), err => {
```

Comment:
It might be a good idea to use built-in [promisify](https://nodejs.org/api/util.html#util_util_promisify_original), assigning an async version to a top-level (somewhere in the beginning of a file) constant. We can do it for `readFile` and other calls too.

```ts
const writeFileAsync = promisify(writeFile);
```

Reply:
`promisify` was added in Node 8, but we still support Node 6, so maybe after we drop it :)

Reply:
Someday we will even be able use `fs` with Promises out-of-the-box:
https://nodejs.org/api/fs.html#fs_fs_promises_api

:)